contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
           URL where to find further material (documentation, examples) relevant
           to the application definition.
       enumeration: [NXopt]
-    experiment_identifier:
+    title:
+    start_time(NX_DATE_TIME):
+      doc: |
+        Datetime of the start of the measurement.
+        Should be a ISO8601 date/time stamp. It is recommended to add an explicit time zone,
+        otherwise, the local time zone is assumed per ISO8601.
+    end_time(NX_DATE_TIME):
+      exists: recommended
+      doc: |
+        Datetime of the end of the measurement.
+        Should be a ISO8601 date/time stamp. It is recommended to add an explicit time zone,
+        otherwise the local time zone is assumed per ISO8601.
+    experiment_identifier(NXidentifier):
+      exists: recommended
       doc: |
         A (globally persistent) unique identifier of the experiment.
-        (i) The identifier is usually defined by the facility or principle
-        investigator.
+        (i) The identifier is usually defined by the facility, laboratory or
+        principal investigator.
         (ii) The identifier enables to link experiments to e.g. proposals.
+      service(NX_CHAR):
+      identifier(NX_CHAR):
+      is_persistent(NX_BOOLEAN):
```

Comment:
Is it strictly neccessary to require these fields? Could just be recommended or even optional.

Reply:
ah yea, was not intended. Just have to remember that the default value is required. Was adjusted for upcomming commit.